xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -1180,36 +1244,33 @@ static void DisplayReconfigured(CGDirectDisplayID display,
     oldRefreshRate = m_refreshRate;
 
     // send a message so that videoresolution (and refreshrate) is changed
-    NSWindow* win = m_appWindow;
-    NSRect frame = win.contentView.frame;
-    CServiceBroker::GetAppMessenger()->PostMsg(TMSG_VIDEORESIZE, frame.size.width,
-                                               frame.size.height);
+    dispatch_sync(dispatch_get_main_queue(), ^{
+      NSWindow* win = m_appWindow;
+      NSRect frame = win.contentView.frame;
+      CServiceBroker::GetAppMessenger()->PostMsg(TMSG_VIDEORESIZE, frame.size.width,
+                                                 frame.size.height);
+    });
+  }
+  // store window position in window mode
+  if (!m_bFullScreen)
+  {
+    dispatch_sync(dispatch_get_main_queue(), ^{
+      settings->SetInt(SETTING_WINDOW_LEFT.UTF8String, m_appWindow.frame.origin.x);
+      settings->SetInt(SETTING_WINDOW_TOP.UTF8String, m_appWindow.frame.origin.y);
+      settings->Save();
+    });
   }
 }
 
 void CWinSystemOSX::WindowChangedScreen()
 {
-  // user has moved the window to a
-  // different screen
-  NSOpenGLContext* context = [NSOpenGLContext currentContext];
-  m_lastDisplayNr = 0;
-
+  // user has moved the window to a different screen
+  NSWindow* window = [[NSApplication sharedApplication] keyWindow];
```

Comment:
I do not see why you are calling `keyWindow` to get the window.  First, it's not guaranteed that the Kodi window is the `keyWindow`. Perhaps you mean `mainWindow`, which is true unless there is an About box open, for example.

Still, I see why you're seeking to replace `context.view.window` (terrible!), but you might look in your instance variables. I think you'll find `m_appWindow` will do nicely.